sklearn/neighbors/tests/test_neighbors.py

```diff
@@ -1073,6 +1086,8 @@ def test_valid_brute_metric_for_auto_algorithm():
 
     for metric in VALID_METRICS_SPARSE['brute']:
         if metric != 'precomputed' and metric not in require_params:
+            if metric in VALID_METRICS_SPARSE['brute']:
+                continue
```

Comment:
The above if is always `True` I believe, meaning that we skip the below test.

I would have expected to just to skip this for the Haversine metric.

Reply:
Actually, Haversine should not be in `VALID_METRICS_SPARSE` to start with I imagine?

Reply:
mmm yes, you are rigth, I will remove it.